src/pybind/mgr/cephadm/module.py

```diff
@@ -1723,6 +1874,15 @@ def _create_daemon(self,
                 if spec.ports:
                     ports.extend(spec.ports)
 
+            if daemon_spec.daemon_type == 'cephadm-exporter':
+                if not reconfig:
+                    assert daemon_spec.host
+                    deploy_ok = self._deploy_cephadm_binary(daemon_spec.host)
+                    if not deploy_ok:
+                        msg = f"Unable to deploy the cephadm binary to {daemon_spec.host}"
+                        self.log.warning(msg)
+                        return msg
```

Comment:
I really made great effort to move all the different `daemon_type == 'cephadm-exporter'` into the `services` module. Seeing those things popping up again is unfortunate. 

Any way you can use proper structure here?

Reply:
I'm not quite sure what you mean..```daemon_type == "X"``` is a common idiom in the code. Can you be more specific if you want further changes?